### PR TITLE
add option to set max lines to save for panes

### PR DESCRIPTION
**Pull request by MLFlexer**
Add `resurrect.set_max_nlines` option to set the max number of lines saved in each pane.

### CODE REVIEW

File: plugin/resurrect/pane_tree.lua [hunk 1]
```diff
@@ -1,5 +1,9 @@
 local wezterm = require("wezterm")
+
+---@class pane_tree_module
+---@field max_nlines integer
 local pub = {}
+pub.max_nlines = 2000
```
Comment (MLFlexer) [inline plugin/resurrect/pane_tree.lua:6]: This is just some arbitrary number, I dont know what would be a good value for this. If there is a suggestion for a better default value, then I am open for this.

Comment (joncrangle) [reply plugin/resurrect/pane_tree.lua]: The updates look great. 

I don't think you need to set a default `max_nlines`, and you can probably just remove this line. 

From my perspective, it's totally fine to use `root.pane:get_dimensions().scrollback_rows` as the default, and then someone can provide their own value in their config with `resurrect.set_max_nlines(number)`.

Comment (MLFlexer) [reply plugin/resurrect/pane_tree.lua]: The only thing I'm worried about is the default performance. 

If a pane is never being removed or cleared for text, then the text from the current session will append to the previous sessions. Doing this over some period of time would reach the `scrollback_rows` limit which could be very large. 
If a user does this for multiple panes, then the saving/loading of the state can become slow over time. Which would result in a bad user experience. Another thing to note is that if the output is colored, then each line becomes a very long string as the escape sequence is also a part of it. 
You can test the limit by running:
```bash
for i in {1..10000}; do
  ls -la
done
```

In the latest commit it is set to 1000 lines which is still a lot of lines, and from my testing it seems like `3554` is the default limit. 
I would even consider making the default even less, as I personally only care about the last ~3 shell outputs, so even 200 lines would probably be plenty. However this is only my usecase and it is hard to know how others use the plugin. What do you think? @joncrangle would be the best middle ground? Or do you not see the same performance problem as I do? I would love to hear your oppinion on this, as I am unsure what the best approach is here.

Comment (joncrangle) [reply plugin/resurrect/pane_tree.lua]: It appears that the Wezterm default is 3500 nlines, so this is probably a good choice as default. 

https://github.com/wez/wezterm/blob/main/config%2Fsrc%2Fconfig.rs#L1637-L1639


The README can specify how to save more if they would like.

Comment (MLFlexer) [reply plugin/resurrect/pane_tree.lua]: Changed to 3500

**Pull request merged**